main.go

```diff
@@ -53,51 +60,54 @@ func main() {
 		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
 	}
 	log := zapr.NewLogger(zapLog)
+	ctx := logr.NewContext(context.Background(), log)
 
-	err = run(log, args)
+	switch {
+	case args.Configuration != nil:
+		err = configurationCommand(ctx, args.Configuration)
+	case args.Registry != nil:
+		err = registryCommand(ctx, args.Registry)
+	}
 	if err != nil {
 		log.Error(err, "")
 		os.Exit(1)
 	}
+
 	log.Info("gracefully shutdown")
 }
 
-func run(log logr.Logger, args *arguments) (err error) {
-	cs, err := pkgkubernetes.GetKubernetesClientset(args.KubeconfigPath)
+func configurationCommand(ctx context.Context, configurationCommand *ConfigurationCmd) error {
+	fs := afero.NewOsFs()
+	err := mirror.AddMirrorConfiguration(ctx, fs, configurationCommand.ContainerdRegistryConfigPath, configurationCommand.Registries, configurationCommand.MirrorRegistries)
 	if err != nil {
 		return err
 	}
-	containerdClient, err := containerd.New(args.ContainerdSock, containerd.WithDefaultNamespace(args.ContainerdNamespace))
-	if err != nil {
-		return fmt.Errorf("could not create containerd client: %w", err)
-	}
-	defer func() {
-		err = errors.Join(err, containerdClient.Close())
-	}()
+	return nil
+}
 
-	ctx := logr.NewContext(context.Background(), log)
+func registryCommand(ctx context.Context, registryCmd *RegistryCmd) (err error) {
+	log := logr.FromContextOrDiscard(ctx)
 	ctx, cancel := signal.NotifyContext(ctx, syscall.SIGTERM)
 	defer cancel()
 	g, ctx := errgroup.WithContext(ctx)
 
-	// It is fine to immediatly write mirror configuration because it should fallback to another nodes Spegel instance.
-	fs := afero.NewOsFs()
-	if args.ContainerdMirrorAdd {
-		err := mirror.AddMirrorConfiguration(ctx, fs, args.ContainerdRegistryConfigPath, args.Registries, args.MirrorRegistries)
-		if err != nil {
-			return err
-		}
+	cs, err := pkgkubernetes.GetKubernetesClientset(registryCmd.KubeconfigPath)
+	if err != nil {
+		return err
 	}
-	if args.ContainerdMirrorRemove {
-		defer func() {
-			err = errors.Join(err, mirror.RemoveMirrorConfiguration(ctx, fs, args.ContainerdRegistryConfigPath, args.Registries))
-		}()
```

Comment:
I understand that we're not mounting it anymore and that gives us a smaller surface in the long running container, but I do think it's nice of us to keep this and the mount since people may want to try it out for a short while and then remove spegel and that can potentially cause issues for them. I vote for keeping it and the mount.

Reply:
I have created #67 to track this feature, as it should be implemented in a separate PR.